src/controllers/AuthLocalController.js

```diff
@@ -0,0 +1,175 @@
+      if (Object.keys(checkUser).length > 0) {
+        const comparePassword = helper.password.compare(password, checkUser.password);
+        if (!comparePassword) {
+          return res.status(status.UNAUTHORIZED).send({
+            message: 'The credentials you provided is incorrect'
+          });
+        }
+        const payload = { id: checkUser.id, role: checkUser.role };
+        const token = helper.tokenGenerator(payload);
+        delete checkUser.password;
+        return res.status(status.OK).send({
+          user: checkUser,
+          token
+        });
+      }
+      return res
+        .status(status.UNAUTHORIZED)
+        .send({ error: `User with ${email} email doesn't exist!!!` });
+    } catch (error) {
+      return res.status(500).send({ error });
+    }
+  }
+
+  /**
+   * @description - methode to update user
+   * @param {object} req
+   * @param {object} res
+   * @returns {object} updated user
+   */
+  static async update(req, res) {
+    try {
+      const { id } = req.params;
+      const { permissions } = req.body;
+      const isUser = await validate.isUser({ id });
+      if (isUser) {
+        const updateUser = await User.update({ permissions: [permissions] }, { id });
+        return updateUser
+          ? res.status(200).send({
+            message: 'user updated successfuly'
+          })
+          : res.status(401).send({ error: 'Unauthorized access' });
+      }
+      return res.status(404).send({ error: 'user not found' });
+    } catch (error) {
+      return error;
+    }
+  }
+
+  /**
+   * @description function to delete user
+   * @param {object} req
+   * @param {object} res
+   * @returns {object} return true if user deleted or false when user not deleted
+   */
+  static async delete(req, res) {
+    try {
+      const { id } = req.params;
+      const findUser = await validate.isUser({ id });
+      if (findUser) {
+        const deleteUser = await User.deleteUser({ id });
+        return deleteUser
+          ? res.status(status.OK).send({ message: 'User account deleted successfully' })
+          : res.status(status.UNAUTHORIZED).send('Unauthorized access');
+      }
+      return res.status(status.NOT_FOUND).send({ error: 'user not found!!!' });
+    } catch (error) {
+      return res.status(status.SERVER_ERROR).send({ error: error.message });
+    }
+  }
+
+  /**
+   * @description methode to find all users
+   * @param {object} req
+   * @param {object} res
+   * @returns {object} return all users
+   */
+  static async getAll(req, res) {
+    try {
+      const { role } = req.body;
+      const users = await User.findAll({ role });
+      delete users.password;
+      return users
+        ? res.status(status.OK).send({ users })
+        : res.status(status.UNAUTHORIZED).send({
+          error: 'Unauthorized access'
+        });
+    } catch (error) {
+      return error;
+    }
+  }
+
+  /**
+   * @description function for admin to create user
+   * @param {object} req
```

Comment:
It's better you go through your codes and descriptive JSDoc comment where applied

Reply:
yes, change applied